buckets/buckets.go

```diff
@@ -412,13 +418,9 @@ Loop:
 }
 
 func (b *Buckets) saveDealsInArchive(ctx context.Context, key string, dbID thread.ID, ffsToken string, c cid.Cid, opts ...Option) error {
-	args := &Options{}
-	for _, opt := range opts {
-		opt(args)
-	}
-	buck := &Bucket{}
-	if err := b.threads.FindByID(ctx, dbID, CollectionName, key, buck, db.WithTxnToken(args.Token)); err != nil {
-		return fmt.Errorf("finding bucket: %s", err)
+	buck, err := b.Get(ctx, dbID, key, opts...)
```

Comment:
Using bucket methods here which are wired to handle adding / updating the collection schema.